content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+
+More specifically, when Alice and Bob call [Split()](http://www.noiseprotocol.org/noise.html#the-symmetricstate-object) in order to derive the two final encryption and decryption Cipher States, 
+they further compute `session-id  = HKDF(h)` using the supported key derivation function `HKDF`.
+
+Such `session-id` will uniquely identify the Noise cryptographic session instantiated on completion of a Noise handshake,
+which would then consists of the tuple `(session-id, CSOutbound, CSInbound)`. 
+
+Once a Noise session is instantiated, 
+any further encrypted message between Alice and Bob within this session is exchanged on a `contentTopic` with name `/wakunoise/1/<ct-id>/proto`, 
+where `ct-id = Hash(Hash(session-id))`.
+
+
+# Session states
+
+A Noise session corresponding to a certain `session-id`:
+- is always **active** as long as it is not marked as **stale**. 
+For an active `session-id`, new messages are published on the content topic `/wakunoise/1/<ct-id>/proto`;  
+- is marked as **stale** if a [session termination message](https://rfc.vac.dev/spec/35/#session-termination-message) containing `Hash(session-id)` is published on the content topic `/wakunoise/1/<ct-id>/proto`. 
+Session information relative to stale sessions  MAY be deleted from users' device, unless required for later channel binding purposes.
+
+When a Noise session is marked as stale, it means that one party requested its termination while being online, 
+since publication of a hash pre-image for `ct-id` is required (i.e. `Hash(session-id)`).
+
+Currently, it is not possible to mark a Noise session as stale when `session-id` is lost or gets corrupted in users' devices.
+However, since `session-id` is shared between Alice and Bob, 
+one party MAY decide to mark a Noise session as stale if no message from the other end was received within a certain fixed time window. 
+
+The above mechanism allows a Noise session to be marked as stale either privately or publicly, 
+depending if `Hash(session-id)` is sent on `/wakunoise/1/<ct-id>/proto` to the other party in encrypted form or not, respectively. 
+
+When a Noise session is publicly marked as stale, 
+network peers MAY discard all [stored](https://rfc.vac.dev/spec/13/) messages addressed to the content topic `/wakunoise/1/<ct-id>/proto`. 
+In this the case and in order for parties to retrieve any eventually delayed message, 
+peers SHOULD wait a fixed amount of time before discarding stored messages corresponding to a stale Noise session. 
+
+A stale Noise session cannot be directly marked as active
+and parties are required to instantiate a new Noise session if they wish to communicate again. 
+
+However, parties can optionally include the `session-id` corresponding to a stale Noise session in the [prologue information](https://noiseprotocol.org/noise.html#prologue) employed in the Noise handshake they execute to instantiate their new Noise session. 
```

Comment:
> [prologue information](https://noiseprotocol.org/noise.html#prologue)

can you please supply a link to our own spec?

Reply:
The term _reactivating the noise session_ is a bit confusing. By using the Noise prologue input, the `session id` is included in the `h` variable of the handshake, and then the parties start the _handshake_ by such `h` value, so how it is relevant to reactivating a _Noise session_?

Reply:
The term "re-activate" is used in this sentence

> This effectively emulates a mechanism to "re-activate" a stale Noise session by binding it to a newly created active Noise session.

and I used double quotes on purpose to stress that such term is only to provide an intuition of what is happening in the technical description that follows.

Here the word "re-activate" made sense to me because I was talking about the scenario where a session remains active for a long period of time but with no message received from the other party (think of a device that remains offline 1 year or got stolen). Then the other party can unilaterally decide to mark such session as stale (one reason I was considering this possibility here, is that in the future store nodes may provide as a service storage of messages for longer period of time than current 30 days). If, by chance, the offline device comes back online later on, and wants to communicate with the other party again (which in the meanwhile marked the session stale), he then need to instantiate a new session with him, but he can do it in a way that the new session is bind to the old staled one, in fact emulating a session _re-activation_.

Reply:
Thanks for the explanation, what is the benefit such re-activating from your PoV? Why not establishing a new one? 
BTW, the session-id must be also persisted by the other party in order to be able to engage in the reactivated session, right?

Reply:
The advantage is to bind new messages to the any previous executed handshake. Imagine if you keep and update the handshake hash for any exchanged message, such binding would be stronger. Of course it has to be persisted in order to be reactivated: clarified it in https://github.com/vacp2p/rfc/pull/504/commits/9cff4db8f1ade87e9dada9cc7eb93f6cce55b13f